File: Data Structures and Libraries/1069 - Diamonds and Sand/1069.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {

        try (Scanner scan = new Scanner(System.in)) {
            int N, i, i2, i3;
            char[] Pilha = new char[2048];
            int diamonds;

            N = Integer.parseInt(scan.nextLine());

            for (i = 0; i < N; i++) {
                String input = scan.nextLine();
                Pilha = input.toCharArray();
                diamonds = 0;

                for (i2 = 0; i2 < Pilha.length; i2++) {
                    if (Pilha[i2] == '<') {
                        for (i3 = i2; i3 < Pilha.length; i3++) {
                            if (Pilha[i3] == '>') {
                                diamonds++;
                                Pilha[i3] = '.';
                                break;
                            }
                        }
                    }
                }
                System.out.println(diamonds);
            }
            scan.close();
        }
    }
}

File: Data Structures and Libraries/1069 - Diamonds and Sand/1069.py
N = int(input())

for _ in range(N):
    input_str = input()
    Pilha = list(input_str)
    diamonds = 0

    for i2 in range(len(Pilha)):
        if Pilha[i2] == '<':
            for i3 in range(i2, len(Pilha)):
                if Pilha[i3] == '>':
                    diamonds += 1
                    Pilha[i3] = '.'
                    break

    print(diamonds)